client/ayon_core/pipeline/stagingdir.py

```diff
@@ -0,0 +1,218 @@
+STAGING_DIR_TEMPLATES = "staging"
+
+
+def get_staging_dir_config(
+    host_name,
+    project_name,
+    task_type,
+    task_name,
+    product_type,
+    product_name,
+    project_settings=None,
+    anatomy=None,
+    log=None,
+):
+    """Get matching staging dir profile.
+
+    Args:
+        host_name (str): Name of host.
+        project_name (str): Name of project.
+        task_type (str): Type of task.
+        task_name (str): Name of task.
+        product_type (str): Type of product.
+        product_name (str): Name of product.
+        project_settings(Dict[str, Any]): Prepared project settings.
+        anatomy (Dict[str, Any])
+        log (Optional[logging.Logger])
+
+    Returns:
+        Dict or None: Data with directory template and is_persistent or None
+
+    Raises:
+        ValueError - if misconfigured template should be used
+
+    """
+    settings = project_settings or get_project_settings(project_name)
+
+    staging_dir_profiles = settings["core"]["tools"]["publish"][
+        "custom_staging_dir_profiles"
+    ]
+
+    if not staging_dir_profiles:
+        return None
+
+    if not log:
+        log = Logger.get_logger("get_staging_dir_config")
+
+    filtering_criteria = {
+        "hosts": host_name,
+        "task_types": task_type,
+        "task_names": task_name,
+        "product_types": product_type,
+        "product_names": product_name,
+    }
+    profile = filter_profiles(
+        staging_dir_profiles, filtering_criteria, logger=log)
+
+    if not profile or not profile["active"]:
+        return None
+
+    if not anatomy:
+        anatomy = Anatomy(project_name)
+
+    # get template from template name
+    template_name = profile["template_name"]
+    _validate_template_name(project_name, template_name, anatomy)
+
+    template = anatomy.templates[STAGING_DIR_TEMPLATES][template_name]
+
+    if not template:
+        # template should always be found either from anatomy or from profile
+        raise ValueError(
+            "Staging dir profile is misconfigured! "
+            "No template was found for profile! "
+            "Check your project settings at: "
+            "'ayon+settings://core/tools/publish/custom_staging_dir_profiles'"
+        )
+
+    data_persistence = profile["custom_staging_dir_persistent"]
+
+    return {"template": template, "persistence": data_persistence}
+
+
+def _validate_template_name(project_name, template_name, anatomy):
+    """Check that staging dir section with appropriate template exist.
+
+    Raises:
+        ValueError - if misconfigured template
+    """
+    if template_name not in anatomy.templates[STAGING_DIR_TEMPLATES]:
+        raise ValueError(
+            (
+                'Anatomy of project "{}" does not have set'
+                ' "{}" template key at Staging Dir section!'
+            ).format(project_name, template_name)
+        )
+
+
+def get_staging_dir_info(
+    host_name,
+    project_entity,
+    folder_entity,
+    task_entity,
```

Comment:
I would move `host_name` after entities.
```suggestion
    project_entity,
    folder_entity,
    task_entity,
    host_name,
```